packages/editor-ui/src/utils/nodeTypesUtils.ts

```diff
@@ -305,3 +313,247 @@ export const hasOnlyListMode = (parameter: INodeProperties): boolean => {
 		parameter.modes[0].name === 'list'
 	);
 };
+
+// A credential type is considered required if it has no dependencies
+// or if it's only dependency is the main authentication fields
+export const isRequiredCredential = (
+	nodeType: INodeTypeDescription | null,
+	credential: INodeCredentialDescription,
+): boolean => {
+	if (!credential.displayOptions || !credential.displayOptions.show) {
+		return true;
+	}
+	const mainAuthField = getMainAuthField(nodeType);
+	if (mainAuthField) {
+		return mainAuthField.name in credential.displayOptions.show;
+	}
+	return false;
+};
+
+// Finds the main authentication filed for the node type
+// It's the field that node's required credential depend on
+export const getMainAuthField = (nodeType: INodeTypeDescription | null): INodeProperties | null => {
+	if (!nodeType) {
+		return null;
+	}
+	const credentialDependencies = getNodeAuthFields(nodeType);
+	const authenticationField =
+		credentialDependencies.find(
+			(prop) =>
+				prop.name === MAIN_AUTH_FIELD_NAME &&
+				!prop.options?.find((option) => option.value === 'none'),
+		) || null;
+	// If there is a field name `authentication`, use it
+	// Otherwise, try to find alternative main auth field
+	const mainAuthFiled =
+		authenticationField || findAlternativeAuthField(nodeType, credentialDependencies);
+	// Main authentication field has to be required
+	const isFieldRequired = mainAuthFiled ? isNodeParameterRequired(nodeType, mainAuthFiled) : false;
+	return mainAuthFiled && isFieldRequired ? mainAuthFiled : null;
+};
+
+// A field is considered main auth filed if:
+// 1. It is a credential dependency
+// 2. If all of it's possible values are used in credential's display options
+const findAlternativeAuthField = (
+	nodeType: INodeTypeDescription,
+	fields: INodeProperties[],
+): INodeProperties | null => {
```

Comment:
```suggestion
) => {
```

Reply:
I find this function a bit hard to follow. To consider:
- `reduce` for `forEach` (since we are reducing the param into `dependentAuthFieldValues`)
- Inverted condition to reduce nesting
- `some` for `find` (since a `find` that returns a boolean is rather unexpected)
- `??` instead of `||`

Reply:
Agree. Leaving this refactor for P1, since we are splitting credentials logic there so this wll also be reworked.